### DISCUSSION TITLE
fill-extrusion-pattern behaviour

### DISCUSSION THREAD

**Discussion by cigone-openindoor** [maplibre/maplibre-gl-js#1090] [Q&A]
When trying to apply a 64x64px pattern in an extruded LineString, with height 1, it's quite clear that how image pattern is applied as a texture in that axe (z).

With a good pow2 image ratio, we could get easily:
- 1 meter height => image appears once
- 2 meters height => image appears twice
![Screenshot 2022-03-13 at 11 37 42](https://user-images.githubusercontent.com/69067574/158055677-e647ff5a-83dc-436e-b397-b107e7b7ea81.png)

However, it's not clear at all how width ratio is applied.

With a 1 meter width in equator (latitude=0), image is cut (not full width).
![Screenshot 2022-03-13 at 11 29 16](https://user-images.githubusercontent.com/69067574/158055457-d7ec3822-8e95-40f1-bec2-96ea05a4f2cd.png)

With a 1 meter width in latitude 45, image appears more than once for the width.
![Screenshot 2022-03-13 at 11 28 39](https://user-images.githubusercontent.com/69067574/158055460-ebd8bf7c-8aab-482d-9e2e-14f26aac7502.png)

And with a power2 image, offset bottom-left suits well, otherwise it's "random".

**Could we have an explanation on the way pattern is applied on an LineString extrusion ?**

My goal is to be able to have my pattern image displayed completely and no more (no repeat, no offset), as a kind of building wall.

This could be very powerful to apply some texture on buildings, or for indoor stuffs...

**Comment by HarelM** [comment]
This feels more of a question than an issue, can I move to discussions?

**Comment by cigone-openindoor** [comment]
Yes
